### PR TITLE
Keep original path and params when redirecting deep links to embed

### PR DESCRIPTION
**Pull request by kirillplatonov**
### What this PR does

Fixes deep link behavior to make sure the original path and params are preserved after redirect to app embed. Right now all deep links are redirected to home page. In addition to deep links it fixes opening links in new tab in app embed.

https://github.com/Shopify/shopify_app/assets/839922/a1c43f6a-8486-40ea-af4c-5ef384906807

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] Update `README.md`, if appropriate.
- [x] Update any relevant pages in `/docs`, if necessary
- [x] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

File: test/controllers/concerns/embedded_app_test.rb [hunk 1]
```diff
@@ -75,15 +75,23 @@ def redirect_to_embed
...
     get :redirect_to_embed, params: { shop: shop }
-    assert_redirected_to "https://#{shop}/admin/apps/#{ShopifyApp.configuration.api_key}"
+    assert_redirected_to "https://#{shop}/admin/apps/#{ShopifyApp.configuration.api_key}/redirect_to_embed"
+  end
+
+  test "#redirect_to_embed_app_in_admin keeps original path and params when redirecting to the embed app" do
+    ShopifyApp.configuration.embedded_app = true
+
+    shop = "my-shop.myshopify.com"
+    get :redirect_to_embed, params: { shop: shop, foo: "bar" }
```
Comment (zzooeeyy) [inline test/controllers/concerns/embedded_app_test.rb:93]: Could we add these extra params to test that it [doesn't get included](https://github.com/Shopify/shopify_app/pull/1869/files#diff-ac5f4b969898cd49605b9d34ccb2d15486a7ede28a99560265795785ca908a81R29) in the redirect?
```suggestion
    get :redirect_to_embed, params: { shop: shop, foo: "bar", host: "host", id_token:"id_token" }
```

**Review by zzooeeyy — APPROVED** [review]
Thanks!! Looks good to me, I just have a small comment on the test case.

File: test/controllers/concerns/embedded_app_test.rb (hunk 1 above)
Comment (kirillplatonov) [reply test/controllers/concerns/embedded_app_test.rb]: Yes, updated

**Review by matteodepalo — APPROVED** [review]

**Comment by kirillplatonov** [comment]
@matteodepalo resolved changelog conflict. Ready for merge

**Pull request merged**